qiskit_machine_learning/neural_networks/opflow_qnn.py

```diff
@@ -134,7 +155,15 @@ def _backward(self, input_data: Optional[np.ndarray], weights: Optional[np.ndarr
 
         # iterate over rows, each row is an element of a batch
         batch_size = input_data.shape[0]
-        grad_all = np.zeros((batch_size, *self.output_shape, self.num_inputs + self.num_weights))
+        if self._input_gradients:
+            grad_all = np.zeros((batch_size,
+                                 *self._output_shape,
+                                 self._num_inputs + self._num_weights))
+        else:
```

Comment:
Pretty minor comment: Since the the body of the if's is so similar, only in one case it has the extra self._num_inputs would it better to set up the third parameter value based on the if and just have the one np.zeros block. Saves someone checking back and forth across the if/else looking for the difference.

Reply:
Agree. Looks better.